### DISCUSSION TITLE
Environment config issue

### DISCUSSION THREAD

**Discussion by ScreamZ** [vercel/vercel#4381] [General] [answered]
Hello,

I have two domains on my app.

One connected to my git branch `master` (default) which point to https://mydomain.com.
One other connected to my `dev` branch, which point to https://staging.mydomain.com.

I'm expecting to see the branch's auto-deploys to run on `mydomain.com` the `production` configuration and on `staging.mydomain.com` the `staging` configuration.

This works well until I push/merge something onto master (or run `vc --prod`), then the production gets deployed, but it also overrides the staging domain with production config… Until I release a new commit on `dev` branch (or `vc`).

Is there any way to get clear separation on both?

Regards

**Comment by styfle** [comment, marked as answer]
Hi @ScreamZ 

The default branch is considered `production` so pushing/merging to that branch will apply the production domain. Also, running `vc --prod` will apply the production domains.

If you set up staging using [Branch Domains](https://vercel.com/blog/branch-domains), then this domain will be assigned when you push/merge to your branch.

If this is not working correctly, please contact support so we can take a close look at your specific account, thanks!

https://vercel.com/support/request
